src/backend/expungeservice/expunger/analyzers/time_analyzer.py

```diff
@@ -133,6 +135,28 @@ def _mark_as_time_ineligible(charges: List[Charge], reason: str, eligibility_dat
         for charge in charges:
             charge.set_time_ineligible(reason, eligibility_date)
 
+    @staticmethod
+    def _set_arrests_as_eligible(charges_with_summary: ChargesWithSummary):
+        # If most recent conviction is time eligible
+        # AND all other charges in the same case are arrests
+        # Then make all other charges in the case time eligible
+        mrc = charges_with_summary.most_recent_conviction
+        second_mrc = charges_with_summary.second_most_recent_conviction
+        if mrc and mrc.expungement_result.time_eligibility and mrc.expungement_result.time_eligibility.status is EligibilityStatus.ELIGIBLE:
+            if not second_mrc:
+                for charge in mrc.case()().charges:
+                    if charge.acquitted():
+                        charge.set_time_eligible()
+        else:
```

Comment:
What happens without this else block?

Reply:
Without the else block, arrests on a case with a second mrc don't have the correct eligibility date

Reply:
Isn't that what line 23/24 of this file is for?

Reply:
I see. I feel like there must be a better way to codify the combination of this and line 23/24 but I think this is fine for now.

Reply:
So based on what I read of 7(b), I think the charges should be set time ineligible until the mrc is 10 years old (not until the second mrc is 10 years old as is written now). @wittejm What do you think?

Reply:
After a bit more thinking, I think this else block is unnecessary. The eligibility for everything that is not the mrc is 10 years after the mrc date, which is taken care of in line 23/24. @wittejm Any input here?

Reply:
If the mrc becomes eligible after the second_mrc is 10 years old, then the arrests on the same case as the mrc fall under this rule, and have the same eligibility. The reason for the `else` here is the same as the reason for having this whole fix: the arrests on this case are not blocked by the mrc, but instead only by the second_mrc

Reply:
Though I think this  actually needs another tweak: the eligibility of the arrests on the same case as the mrc get pushed out by 10 years from the date of the second_mrc, and also by 3 years from the date of the mrc, whichever is farther out.

So for example if the mrc is 1 year old and the second_mrc is 9 years old, the arrests on the mrc's case become eligible in 2 years, not 1. 

It's confusing as hell. But I'm pretty sure there's a test case here that would be failing given the current logic.

Reply:
Ah yes, I was thinking about 7(b) too strictly again. Here my high-level overview of the TimeAnalyzer without Class B Felonies as of now.

```
if mrc and second_mrc:
    # mrc and arrests in same case is eligible the max of mrc + 3 years and second_mrc + 10 years
    # everything else is eligible mrc + 10 years
elif mrc:
    # mrc and arrests in same case is eligible mrc + 3 years
    # everything else is eligible mrc + 10 years
elif mrd and second_mrd:
    # mrd is eligible second_mrd + 3 years
    # everything else is eligible mrd + 3 years
elif mrd:
    # mrd is eligible now
else:
    # no recent convictions or dismissals
```

Reply:
The code also needs a check to see whether `mrc.case != second_mrc.case`. It feels like this logic should be grouped together with `_check_mrc_time_eligibility`, otherwise there will be a lot of duplicated code.

Reply:
I think refactoring this whole thing is probably a good idea, but let's definitely get this fix in so we can include it in a new version push this week. 

While at it there's probably a way to overhaul the way violations are handled with the mrc/second_mrc so we don't need edge case fixes for these new/old/old bugs: #718 #681 #654